daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/ValueCoder.scala

```diff
@@ -156,12 +158,39 @@ object ValueCoder {
       value <- decodeValue(decodeCid, version, protoValue0.getValue)
     } yield VersionedValue(version, value)
 
+  // We need (3 * MAXIMUM_NESTING + 1) as record and maps use :
+  // - 3 messages for the cases of non empty records/maps)
+  // - 2 messages for the cases of empty records/maps
+  // Note the number of recursions is one less than the number of nested messages.
+  private[this] val MAXIMUM_PROTO_RECURSION_LIMIT = 3 * MAXIMUM_NESTING + 1
+
   def decodeValue[Cid](
       decodeCid: DecodeCid[Cid],
       protoValue0: proto.VersionedValue,
   ): Either[DecodeError, Value[Cid]] =
     decodeVersionedValue(decodeCid, protoValue0) map (_.value)
 
+  def parseValue(bytes: ByteString): Either[DecodeError, proto.Value] =
+    Try {
+      val cos = CodedInputStream.newInstance(bytes.asReadOnlyByteBuffer())
```

Comment:
Using short name for local variable is in my opinion better, specially when the scope of the variable is very short.  

I quote https://docs.scala-lang.org/style/naming-conventions.html: 

> Because of Scala’s roots in the functional languages, it is quite normal for local names to be very short:
>
> def add(a: Int, b: Int) = a + b
>
>This would be bad practice in languages like Java, but it is good practice in Scala. This convention works because properly-written Scala methods are quite short, only spanning a single expression and rarely going beyond a few lines. Few local names are used (including parameters), and so there is no need to contrive long, descriptive names. This convention substantially improves the brevity of most Scala sources. **This in turn improves readability, as most expressions fit in one line and the arguments to methods have descriptive type names.**
>
> This convention only applies to parameters of very simple methods (and local fields for very simply classes); everything in the public interface should be descriptive. Also note that the names of arguments are now part of the public API of a class, since users can use named parameters in method calls.